Instance volume configuration through disk device (#12089)
canonical/lxd: doc/reference/devices_disk.md

      lxc config device add <instance_name> <device_name> disk source=cloud-init:config

> ## Initial Configuration for Instance Root Disk Devices

Reviewer: ```suggestion
## Initial configuration for instance root disk devices
```

Initial configuration allows setting specific configurations for the root disk devices of new instances.
These settings are prefixed with `initial.` and are only applied when the instance is created.

> This enables the creation of instances with unique configurations, independent of the default storage pool settings.

Reviewer: ```suggestion
This allows creating instances that have unique configurations, independent of the default storage pool settings.
```

> For example, initial configuration for `block.filesystem` can be set to an existing profile, taking effect for each new instance created:

Reviewer: ```suggestion
For example, you can add an initial configuration for `block.filesystem` to an existing profile, and this will then take effect for each new instance you create using this profile:
```

    lxc profile device set <profile_name> <device_name> initial.block.filesystem=btrfs

> Initial configuration can also be set directly when an instance is being created. For example:

Reviewer: ```suggestion
You can also set an initial configuration directly when creating an instance. For example:
```

      lxc config device add <instance_name> <device_name> disk source=cloud-init:config

> ## Initial configuration for instance root disk devices

Reviewer 2: Can we call this `Initial volume configuration for instance root disk devices` and then for all places that we say `Initial configuration` change to `Initial volume configuration`.

Initial configuration allows setting specific configurations for the root disk devices of new instances.
These settings are prefixed with `initial.` and are only applied when the instance is created.
This allows creating instances that have unique configurations, independent of the default storage pool settings.

For example, you can add an initial configuration for `block.filesystem` to an existing profile, and this will then
take effect for each new instance you create using this profile:

    lxc profile device set <profile_name> <device_name> initial.block.filesystem=btrfs

> You can also set an initial configuration directly when creating an instance. For example:

Reviewer 2: Could you also add an example of enabling zfs block mode in this section as that is what the original request was for.

Thanks

      lxc config device add <instance_name> <device_name> disk source=cloud-init:config

> ## Initial volume configuration for instance root disk devices

Reviewer: ```suggestion
(devices-disk-initial-config)=
## Initial volume configuration for instance root disk devices
```

Initial volume configuration allows setting specific configurations for the root disk devices of new instances.
These settings are prefixed with `initial.` and are only applied when the instance is created.

> This allows creating instances that have unique configurations, independent of the default storage pool settings.

Reviewer: ```suggestion
This method allows creating instances that have unique configurations, independent of the default storage pool settings.
```

`boot.priority`     | integer   | -         | no        | Boot priority for VMs (higher value boots first)
`ceph.cluster_name` | string    | `ceph`    | no        | The cluster name of the Ceph cluster (required for Ceph or CephFS sources)
`ceph.user_name`    | string    | `admin`   | no        | The user name of the Ceph cluster (required for Ceph or CephFS sources)

> `initial.*`         |           |           | no        | [Initial volume configuration for instance root disk devices](#Initial-volume-configuration-for-instance-root-disk-devices) that allows setting unique configurations independent of default storage pool settings.

Author: Ok, it seems this is absolutely an issue. So my question is, what is the practice to reference a section on the same page?

> `initial.*`         |           |           | no        | [Initial volume configuration for instance root disk devices](#initial-volume-configuration-for-instance-root-disk-devices) that allows setting unique configurations independent of default storage pool settings.

Reviewer: ```suggestion
`initial.*`         |           |           | no        | {ref}`devices-disk-initial-config` that allows setting unique configurations independent of default storage pool settings
```

> 

Reviewer: No need for an empty line here.

`boot.priority`     | integer   | -         | no        | Boot priority for VMs (higher value boots first)
`ceph.cluster_name` | string    | `ceph`    | no        | The cluster name of the Ceph cluster (required for Ceph or CephFS sources)
`ceph.user_name`    | string    | `admin`   | no        | The user name of the Ceph cluster (required for Ceph or CephFS sources)

> `initial.*`         |           |           | no        | {ref}`devices-disk-initial-config` that allows setting unique configurations independent of default storage pool settings

Reviewer: ```suggestion
`initial.*`         | n/a       | -         | no        | {ref}`devices-disk-initial-config` that allows setting unique configurations independent of default storage pool settings
```
